### PR TITLE
Added dark mode (experimental)

### PR DESCRIPTION
**Pull request by tyron**
Experimentation with Dark Mode for webCoRE UI. Some tweaks are still needed, specially on overlay and dialog windows. This is a draft to get feedback as well.

Current implementation detects the system-wide preference from user and uses as default, until the user switches manually. From that on, the user-preferred will be honoured.

I used CSS custom variables extensively for the switch, which means support for webCoRE would have to be dropped on older browsers (https://caniuse.com/css-variables). Not even the old CSS would work on those browsers.

For webCoRE logo and text marks, I just assumed a color that looked nice. Please let me know if there's an official layout to be used over dark backgrounds. I basically updated the SVG files from `#333` to `#CCC` while keeping the blues. In the end, I duplicated the SVG files (adding a suffix for `-light.svg` and `-dark.svg`) but also kept the old files in-place, in case they are being referenced elsewhere (like in the forums). If that's not needed, we can remove the original SVG files.

Some screenshots for reference below. The colors on the editor are based on Solarized theme. I kept the colors from light theme as the originals, but we could reconcile them in the future if we want -- one of the benefits of Solarized is that you can keep all the accent colors and just invert their `base` colors and it will still be readable. 

Screen Shot 2020-12-17 at 4 47 10 PM
Screen Shot 2020-12-17 at 5 03 34 PM
Screen Shot 2020-12-17 at 5 01 36 PM

### CODE REVIEW

File: dashboard/css/modules/dashboard.module.css [hunk 1]
```diff
@@ -470,7 +459,9 @@ location-mode {
...
...az2VRAd3d38vLyojMyHA6185nfeAZhxvIO7yIDWciM4m6hCGCpTqej1szlcnJ4eKigAIhyGysyhrG8c39/rzKQVSwWVTY6YieAwqenJ7UaLrC9va3f42rIQiay0cF329kIJIDFBoOB+jJ+fXl5qdaL6rMmHbgastGBW6HTRocvAQTgs/hqrVbTKU8yLCIbHehCJ7qDSBgJ8CIhD0GEwjhdJqg9Pz/LxsaG6gaDn9GMBACM2yAoKniMENXV0IluMDw+PoYjQCSo1+vqj4S3KK6ADxPrcYeLiwv5/PwMLQfdYGBhm6LTFAEsRjgjIrCobH2e9wCJD5N1CY2Li4vjjjUhxWK1nRXGgQEsYJqFZYKAk6SI84Q1oo2txW9vb+X4+Fj91g3c3QFCmQER23gPYbCACeN4yU8QAAhZcWlpSX0wyFI8J8OSjPyAe3uhUNBZstXBODCRwb2zMEEAMNQtAArKpBRk5+fnsrq6ag3c26mJKPBs6qGjoyPFBkYjAWoZLEl5YGoOcEJrVODeTuI6OzvzrVLBBDYwGgkQNYg+Jt9nkTLGu0BtOv7v9xwXQbYpWoEJbIyZSQD2TOksP3MaSSUuq5s6OkyBgvUJRreBxwR4ESv5pW/iedIE0OFX1oCRKDZFgMXEgvTz/zQJ0AijYATrFAHiM7GWiPBbCYANjGCdIMD0kK4JU37JK20CJDUwgtVx8zEBZwP+8fHxawmADYyUJRMEWOFZI+BEyvEMcPTB9DBNmXMhPlCbZGURg3UqCrXb7WyH0cwnMptSIk0CgaWETTGXJoHAYs4pp9lsmNZBmgTw/8ByOmhDc3V1lTgBdETe0OBnrAHTlpJEcnBwME56cXdke/OQd0vp3UtPbeqZKg5a2UjPSmoISLKbNvVgCtzUO7Nwenoa+lglieY+VgFT4LGK28qk66gHW3G1SAdbTuM4j2M9TtbmeS7qPlpENxv+0EeLf+Jw1yFxfX09Pl6Peg0U5kwVHbEcr3tnYt4XHOj88QXHPK+YdnZ2VDb7Enw+tiumP3HJN+saiKxIaqd2osiiUsRnIcTGA5cgc9P5zG/ONStjeQc/Rwayol5f/fiim+IqUxfdfoe+LDx2dmz58GX3Xw12d3f1Wb/fT/+vBmHcLTN/9kir/QNEM2v58U1SAAAAAABJRU5ErkJggg==) center top no-repeat; */
```
Comment (tyron) [inline dashboard/css/modules/dashboard.module.css:462]:
```suggestion
```

File: dashboard/css/app.css [hunk 2]
```diff
@@ -1272,4 +1375,19 @@ li[instance] {
...
+.table-striped>tbody>tr:nth-of-type(odd) {
+  background-color: var(--table-odd-background);
+}
+
+.table-striped>tbody>tr:nth-of-type(even) {
+  background-color: var(--table-even-background);
+}
+
+.table>tbody>tr>td, .table>tbody>tr>th, .table>tfoot>tr>td, .table>tfoot>tr>th, .table>thead>tr>td, .table>thead>tr>th {
+padding: 0.750em;
```
Comment (tyron) [inline dashboard/css/app.css:1390]:
```suggestion
```

**Comment by idpaterson** [comment]
Well done! Once I have time to do a quick code review I think the best way to move forward with this is to publish it on the staging dashboard so that people can try it out. So much to love about CSS variables, but we'll have to see if @ady624 can check analytics to weigh the impact of breaking the styles in older browsers. 🤞

**Comment by tyron** [comment]
Thanks @idpaterson, that sounds good. According to https://readium.org/readium-css/docs/CSS07-variables.html#fallback, it shouldn't be too hard to define a fallback value for IE11 (without support of themes). It gets a bit ugly though, so I'd leave it to @ady624 to weight in if we need to support older browsers. I feel like webCoRE user base is probably more on devs, and it's hard to think about one still using IE these days...

**Comment by idpaterson** [comment]
The code looks great! I haven't merged the branch yet since there may be some touch-up work but dark mode is available for anyone interested to preview at https://staging.webcore.co/

**Comment by tyron** [comment]
Great, thanks. For sure there's opportunity to improve; I'm not a designer nor have much experience with UX, so the contrast on some colors might not be ideal, specially in an Accessibility mindset. 

I know Modals are the piece with most room for improvements; the dropdown menu hasn't been adapted, for example. I've seen newer versions of that component have support for light/dark modes already (starting with Bootstrap 4), but I tried bumping the version and it was a disaster, so it might be a bigger effort.

**Comment by imnotbob** [comment]
Nice.

There are a few drop downs that come back light/faded, but very nice!

**Pull request merged**